internal/server/models/encryption_key.go

```diff
@@ -3,7 +3,11 @@ package models
 type EncryptionKey struct {
 	Model
 
-	KeyID     int32 `gorm:"uniqueIndex:idx_encryption_keys_key_id"` // a short identifier for the key that can be embedded with the encrypted payload
+	// KeyID is a short identifier for the key that can be embedded with the
+	// encrypted payload.
+	// TODO: remove this field? it's not used anywhere
+	KeyID int32
```

Comment:
I think this is supposed to allow us to support data key rotation, but it is not currently being used from what I can tell.  Nothing reads this field.

There's a keyID in the encrypted payload, but that comes from first 4 bytes of the checksum of the encrypted value.

Reply:
I would like to keep this as you're probably not going to be able to rotate keys without it. Would remove the comment.

Reply:
Right now this field is not used, so I think it's important we at least document that.

When we looked at the code last week, I think we found that we are using a checksum of the encrypted data key to identify the key (not this field). https://github.com/infrahq/secrets/blob/1b971b9e367bb9d8e9837fc488028495a27885ce/secrets.go#L181-L187

So if we want to rotate keys, we'd use that checksum, not this value. If we wanted to use this keyID we'd have to migrate the existing encrypted payloads to use this KeyID first.

Alternatively we could continue to use the checksum of the encrypted data key as the identifier and remove this field.  

I think both are fine options, but we should clarify this in the code so that we don't continue to assume this is the keyID being used.